easybuild/easyblocks/s/score_p.py

```diff
@@ -38,49 +39,83 @@
+                toolchain.GCC: 'gcc',
+                toolchain.IBMCOMP: 'ibm',
+                toolchain.INTELCOMP: 'intel',
+                toolchain.PGI: 'pgi',
+            }
+            comp_fam = self.toolchain.comp_family()
+            if comp_fam in comp_opts:
+                self.cfg.update('configopts', "--with-nocross-compiler-suite=%s" % comp_opts[comp_fam])
             else:
-                raise EasyBuildError("MPI family %s not supported yet (only: %s)",
-                                     mpi_fam, ', '.join(mpi_opts.keys()))
+                raise EasyBuildError("Compiler family %s not supported yet (only: %s)",
+                                     comp_fam, ', '.join(comp_opts.keys()))
+
+            # --with-mpi=(bullxmpi|hp|ibmpoe|intel|intel2|intelpoe|lam|mpibull2|mpich|mpich2|mpich3|openmpi|
+            #             platform|scali|sgimpt|sun)
+            #
+            # Notes:
+            #   - intel:    Intel MPI v1.x (ancient & unsupported)
+            #   - intel2:   Intel MPI v2.x and higher
+            #   - intelpoe: IBM POE MPI for Intel platforms
+            #   - mpich:    MPICH v1.x (ancient & unsupported)
+            #   - mpich2:   MPICH2 v1.x
+            #   - mpich3:   MPICH v3.x & MVAPICH2
+            #               This setting actually only affects options passed to the MPI (Fortran) compiler wrappers.
+            #               And since MPICH v3.x-compatible options were already supported in MVAPICH2 v1.7, it is
+            #               safe to use 'mpich3' for all supported versions although MVAPICH2 is based on MPICH v3.x
+            #               only since v1.9b.
+            #
+            # With minimal toolchains, packages using this easyblock may be built with a non-MPI toolchain (e.g., OTF2).
+            # In this case, skip passing the '--with-mpi' option.
+            mpi_opts = {
+                toolchain.INTELMPI: 'intel2',
+                toolchain.OPENMPI: 'openmpi',
+                toolchain.MPICH: 'mpich3',     # In EB terms, MPICH means MPICH 3.x
+                toolchain.MPICH2: 'mpich2',
+                toolchain.MVAPICH2: 'mpich3',
+            }
+            mpi_fam = self.toolchain.mpi_family()
+            if mpi_fam is not None:
+                if mpi_fam in mpi_opts:
+                    self.cfg.update('configopts', "--with-mpi=%s" % mpi_opts[mpi_fam])
+                else:
+                    raise EasyBuildError("MPI family %s not supported yet (only: %s)",
+                                         mpi_fam, ', '.join(mpi_opts.keys()))
 
-        # auto-detection for dependencies mostly works fine, but hard specify paths anyway to have full control
+        # Auto-detection for dependencies mostly works fine, but hard specify paths anyway to have full control
         deps = {
-            'binutils': ['--with-libbfd=%%s/%s' % get_software_libdir('binutils', fs=['libbfd.a'])],
+            'binutils': ['--with-libbfd=%s'],
```

Comment:
@boegel: There was no good reason; it simply was a bug that went by unnoticed for quite some time.  I've checked all Score-P releases back to v1.1 (Oct 2012) and the `configure` option always worked like this:
```
  --with-libbfd=(yes|no|<Path to libbfd installation>)
                          If you want to build with libbfd support but do not
                          have a libbfd in a standard location, you need to
                          explicitly specify the directory where it is
                          installed. On non-cross-compile systems we search
                          the system include and lib paths per default [yes];
                          on cross-compile systems, however, you have to
                          specify a path [no]. --with-libbfd is a shorthand
                          for --with-libbfd-include=<Path/include> and
                          --with-libbfd-lib=<Path/lib>. If these shorthand
                          assumptions are not correct, you can use the
                          explicit include and lib options directly.
  --with-libbfd-include=<Path to libbfd headers>
  --with-libbfd-lib=<Path to libbfd libraries>
```
That is, some absolute path of libbfd.a was passed in, the configure check tried to use it as a prefix for 'include' and 'lib', this obviously didn't help in making the compile/link check pass, and the compiler adapter fell back to using `nm` instead of libbfd. `configure` prints a big fat warning that the provided path does not exist, but it is so well hidden in the EB log file that basically nobody finds it...